docs/user-guides/client-api-config.md

```diff
@@ -0,0 +1,676 @@
+  int64 timestamp = 3;
+}
+```
+
+<details><summary>Update Configuration Sample (Go)</summary><br>
+
+```go
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/vdaas/vald-client-go/v1/payload"
+	"github.com/vdaas/vald-client-go/v1/vald"
+	"google.golang.org/grpc"
+)
+
+func example() {
+	// Create connection
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	target := "localhost:8080"
+	conn, err := grpc.DialContext(ctx, target)
+	if err != nil {
+		panic(err)
+	}
+	// Init Vald client
+	client := vald.NewValdClient(conn)
+
+	// Update sample
+	location, err := client.Update(ctx, &payload.Update_Request{
+		// Vector information (mandatory)
+		Vector: &payload.Object_Vector{
+			// Set the vector
+			Vector: []float32{0, 1, 2},
+			// Set the specific ID for the vector
+                        // The ID must be already indexed.
+			Id:     "sample",
+		},
+		// Update configuration (optional)
+		Config: &payload.Update_Config{
+			SkipStrictExistCheck: false,
+			Filters: &payload.Filter_Config{
+				Targets: []*payload.Filter_Target{
+					{
+						Host: "vald-ingress-filter",
+						Port: 8081,
+					},
+				},
+			},
+			Timestamp: time.Now().UnixMilli(),
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+	...
+}
+```
+
+</details>
+
+#### skip_strict_exist_check
+
+`skip_strict_exist_check` (default value is `false`) is the flag for checking whether the same set of the vector and ID is already inserted or not.
+If it is set as `true`, the checking function will be skipped.<BR>
+
+When `skip_strict_exist_check` is `false`, the following checking steps will run in the update process:
+
+1. Check whether the set of ID and vector is already inserted.
+   If there is no data, the update process ends with returning the `NOT_FOUND` error.
+1. After passing the step.1, check whether the request vector is the same as the indexed vector.
+   If it is the same, the update process ends with returning the `ALREADY_EXIST` error.
+
+The update process will continue if all of the above steps have been passed.
+
+#### filters
+
+`filters` is the configuration when using filter functions.
+In the `Update` section, it is popular to use ingress filtering.
+
+The detailed configuration is following.
+
+```rpc
+// Filter related messages.
+message Filter {
+
+  // Represent the target filter server.
+  message Target {
+    // The target filter component hostname.
+    string host = 1;
+    // The target filter component port.
+    uint32 port = 2;
+  }
+
+  // Represent filter configuration.
+  message Config {
+    // Represent the filter target configuration.
+    repeated Target targets = 1;
+  }
+}
+```
+
+#### timestamp
+
+`timestamp` is the timestamp when the vector is updated.
+When `timestamp` is not set, the current time will be used.
+
+## Upsert Service
+
+`Upsert` is updating existing vectors in the `vald-agent` or inserting new vectors into the `vald-agent` if the request vector is not indexed.
```

Comment:
```suggestion
`Upsert` service allow user to update existing vectors in the `vald-agent` or inserting new vectors into the `vald-agent` if the request vector is not indexed.
```

Reply:
same as above, do we need to specify vald-agent here?

Reply:
```suggestion
`Upsert` service allows the user to update existing vectors in the Vald cluster or insert new vector(s) if the request vector is not indexed.
```